packages/components/package.json

```diff
@@ -25,26 +26,28 @@
     "scripts": {
         "prebuild": "yarn clean",
         "build": "run-s build:* copy-files",
-        "build:js": "tsc --build ./tsconfig.build.json",
+        "build:js": "tsup",
+        "build:types": "tsc --build ./tsconfig.build.json",
         "copy-files": "copyfiles -u 1 \"src/**/*.css\" dist",
         "clean": "rimraf dist tsconfig.build.tsbuildinfo"
     },
     "peerDependencies": {
-        "react": "^18.0.0",
-        "react-dom": "^18.0.0",
         "@hopper-ui/styled-system": "*",
-        "react-aria-components": "*"
+        "react": "^18.0.0",
+        "react-aria-components": "*",
+        "react-dom": "^18.0.0"
     },
     "dependencies": {
         "@hopper-ui/icons": "^2.1.0",
-        "@popperjs/core": "2.11.2",
-        "react-is": "17.0.2",
-        "text-mask-core": "5.1.2",
-        "use-debounce": "7.0.1"
+        "react-aria-components": "^1.1.1",
+        "react-is": "17.0.2"
     },
     "devDependencies": {
         "@hopper-ui/styled-system": "^2.2.0",
-        "react-aria-components": "^1.1.1"
+        "@popperjs/core": "2.11.2",
```

Comment:
you moved 3 dependencies to dev dependencies, which i think will break code for consumers? use-debounce, text-mask-code and popperjs must be shipped with client code

Reply:
just saw that you added them to noExternal. Can you give me a quick TLDR of what is happening here plz?

Reply:
> If you are using tsup to build for Node.js applications/APIs, usually bundling dependencies is not needed, and it can even break things, for instance, while outputting to [ESM](https://nodejs.org/api/esm.html).
> 
> tsup automatically excludes packages specified in the dependencies and peerDependencies fields in the package.json, but if it somehow doesn't exclude some packages, this library also has a special executable tsup-node that automatically skips bundling any Node.js package.
> 
> tsup-node src/index.ts
> All other CLI flags still apply to this command. You can still use the noExternal option to reinclude packages in the bundle, for example packages that belong to a local monorepository.
> 
> If the regular tsup command doesn't work for you, please submit an issue with a link to your repo so we can make the default command better.

https://tsup.egoist.dev/#excluding-all-packages

From my understanding, they should still be in dependencies, and no need to be in the noExternal flag, as they are not bundled by default.

Reply:
`@popperjs/core`, `use-debounce` and `text-mask-core` do not have proper ESM/CJS syntax or did not have the `"type": "module"` field declared in their package.json. This made the code not work properly during SSR when running in Node.js.

Pre-bundling these dependencies fixes the ESM/CJS issues. Since those are pre-bundled, the consumers don't need to download the dependencies anymore, so we can move them from `dependencies` to `devDependencies`.

I'm looking into it, there might be an alternative. ⌛

Reply:
After doing some tests, here are my results:

|Framework|Bundles dependencies during SSR|Supports `exports`/`import` token without explicit `.mjs` or `"type": "modules"` during SSR|Supports `.css` imports in JavaScript files during SSR|
|---|:-:|:-:|:-:|
|**Next.js**|❌(AFAIK)|❌|❌|
|**Remix (Vite plugin)**|⚠️ Can opt-in and compile incompatible ESM modules with `ssr.noExternal`, but still fails to compile CJS modules|❌|❌|

So yeah... pretty bad support for SSR accross the board because of these dependencies.

I am more convinced that pre-bundling these dependencies is the right choice.